app/src/ui/dispatcher/dispatcher.ts

```diff
@@ -3509,57 +3456,74 @@ export class Dispatcher {
    * - refreshes repo (so changes appear in history)
    * - sets success banner
    * - end operation state
-   * TODO: record successful squash stats
    */
   private async completeMultiCommitOperation(
-    kind: MultiCommitOperationKind,
     repository: Repository,
     count: number
   ): Promise<void> {
     this.closePopup()
 
-    let bannerType: BannerType
+    const {
+      branchesState: { tip },
+      multiCommitOperationState: mcos,
+    } = this.repositoryStateManager.get(repository)
+
+    if (mcos === null) {
+      log.error(
+        '[completeMultiCommitOperation] - No multi commit operation to complete.'
+      )
+      return
+    }
+
+    const { operationDetail, targetBranch, originalBranchTip } = mcos
+    const { kind } = operationDetail
 
+    let banner: Banner
     switch (kind) {
       case MultiCommitOperationKind.Squash:
-        bannerType = BannerType.SuccessfulSquash
+        banner = {
+          type: BannerType.SuccessfulSquash,
+          count,
+          onUndo: () => {
+            this.undoMultiCommitOperation(mcos, repository, count)
+          },
+        }
         break
       case MultiCommitOperationKind.Reorder:
-        bannerType = BannerType.SuccessfulReorder
+        banner = {
+          type: BannerType.SuccessfulReorder,
+          count,
+          onUndo: () => {
+            this.undoMultiCommitOperation(mcos, repository, count)
+          },
+        }
         break
-      case MultiCommitOperationKind.Rebase:
       case MultiCommitOperationKind.CherryPick:
+        const targetBranchName = targetBranch !== null ? targetBranch.name : ''
+        banner = {
+          type: BannerType.SuccessfulCherryPick,
+          targetBranchName,
+          countCherryPicked: count,
```

Comment:
Maybe I'm missing something, but wouldn't be simpler to rename `countCherryPicked` to `count` and leave the old code here? 🤔

Reply:
Cherry-pick still has a different form because it requires the targetBranchName so I didn't see it worth bothering to change.

Reply:
Ouch, didn't notice that, sorry!